sdl-core/src/main/scala/io/smartdatalake/workflow/action/spark/customlogic/CustomDsTransformer.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Smart Data Lake - Build your data lake the smart way.
+ *
+ * Copyright © 2019-2022 ELCA Informatique SA (<https://www.elca.ch>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.smartdatalake.workflow.action.spark.customlogic
+
+
+import io.smartdatalake.util.hdfs.PartitionValues
+import org.apache.spark.sql.{DataFrame, Dataset, Encoder, SparkSession}
+
+import scala.reflect.runtime.universe.TypeTag
+
+/**
+ * Interface to define a custom Spark-Dataset transformation (1:1)
+ * When you implement this interface, you need to provide two case classes: One for your input Dataset
+ * and one for your output Dataset.
+ */
+trait CustomDsTransformer[In <: scala.Product, Out <: scala.Product] extends Serializable {
+  implicit def outputEncoder(implicit typeTag: TypeTag[Out]): Encoder[Out] = org.apache.spark.sql.Encoders.product[Out]
+
+  /**
+   * Function to be implemented to define the transformation between an input and output DataFrame (1:1)
+   *
+   * @param session      Spark Session
+   * @param options      Options specified in the configuration for this transformation
+   * @param inputDS      Input Dataset
+   * @param dataObjectId Id of DataObject of SubFeed
+   * @return Transformed DataFrame
+   */
+  def transform(session: SparkSession, options: Map[String, String], inputDS: Dataset[In], dataObjectId: String): Dataset[Out]
+
+  private[smartdatalake] def transformWithTypeConversion(session: SparkSession, options: Map[String, String], inputDf: DataFrame, dataObjectId: String)(implicit typeTag: TypeTag[In]): DataFrame = {
+    val inputDSEncoder = org.apache.spark.sql.Encoders.product[In]
+    transform(session, options, inputDf.as(inputDSEncoder), dataObjectId).toDF
```

Comment:
Here you should be able to write "inputDf.as[In]"

Reply:
Yes, I could change this code to 
```suggestion
      implicit val inputDSEncoder = org.apache.spark.sql.Encoders.product[In]
    transform(session, options, inputDf.as[In], dataObjectId).toDF
```
But I think it's just less clear that way. I see the point of implicits if it were used in many places, but if it's just this one?